app/bundles/CoreBundle/ErrorHandler/ErrorHandler.php

```diff
@@ -491,18 +491,46 @@ function ($matches) {
                 if (!empty($error['showExceptionMessage'])) {
                     $message = $error['message'];
                 } else {
-                    $message    = 'The site is currently offline due to encountering an error. If the problem persists, please contact the system administrator.';
-                    $submessage = 'System administrators, check server logs for errors.';
+                    unset($error);
+                    $error['message']    = 'The site is currently offline due to encountering an error. If the problem persists, please contact the system administrator.';
+                    $error['submessage'] = 'System administrators, check server logs for errors.';
                 }
-                unset($error);
             }
 
             defined('MAUTIC_OFFLINE') or define('MAUTIC_OFFLINE', 1);
 
             try {
-                ob_start();
-                include __DIR__.'/../../../../offline.php';
-                $content = ob_get_clean();
+                // Get the URLs base path
+                $inDev = false !== strpos($_SERVER['SCRIPT_NAME'], 'index_dev.php');
+                $base  = str_replace(['index.php', 'index_dev.php'], '', $_SERVER['SCRIPT_NAME']);
+
+                // Determine if there is an asset prefix
+                $root = self::$root;
+
+                /** @var array<string, mixed> $paths */
+                $paths = [];
+                include self::$root.'/app/config/paths.php';
+
+                $assetPrefix = $paths['asset_prefix'];
+                if (!empty($assetPrefix)) {
+                    if ('/' == substr($assetPrefix, -1)) {
+                        $assetPrefix = substr($assetPrefix, 0, -1);
+                    }
+                }
+                $assetBase          = $assetPrefix.$base.$paths['assets'];
+                $error['assetBase'] = $assetBase;
+
+                // Allow a custom error page
+                $loader             = new \Twig\Loader\FilesystemLoader(['app/bundles/CoreBundle/Resources/views/Offline', 'app/bundles/CoreBundle/Resources/views/Exception']);
+                $twig               = new \Twig\Environment($loader);
+                $exceptionExtension = new \Mautic\CoreBundle\Templating\Twig\Extension\ExceptionExtension();
+                $twig->addFunction($exceptionExtension->getFunctions()[0]); // Return RootPath
```

Comment:
@escopecz @LordRembo I registered one filter manually, but I do not like I am using `getFunctions()[0]` as it will break if the index changes. For now it seems to be working, if you have a better solution I am glad to hear it

Reply:
If you need that rootPath only for use in a Twig template, isn't it possible to just register the Extension and then call the 'getRoot' function  in the Offline Twig template?
If that's not possible, I would skip using the Extension and just copy the php code that's in there. Sure, it's a bit of duplication but it's a simple bit of code.
`$root = realpath(__DIR__.'/../../../../../../');`

Reply:
How about to send the root path to the template as param? In the second param of the `$twig->redner()` method I mean.

Reply:
@escopecz I already tried that option, it did not work as `traces.html.twig` is being used by Exceptionxxx.html.twig and they do not pass the `root` value.

I did what @LordRembo suggested 

```
                // Allow a custom error page
                $loader             = new \Twig\Loader\FilesystemLoader(['app/bundles/CoreBundle/Resources/views/Offline', 'app/bundles/CoreBundle/Resources/views/Exception']);
                $twig               = new \Twig\Environment($loader);
                // This is the same filter Located at Mautic\CoreBundle\Templating\Twig\Extension\ExceptionExtension;
                $twig->addFunction(new \Twig\TwigFunction('getRootPath', function () {
                    $root = realpath(__DIR__.'/../../../../');
                    return $root;
                }));
```